### ISSUE TITLE
Subscribers and number of videos on YT channel should be presented Dynamically without Hardcoding

### ISSUE THREAD

**Issue by Satyxm** [piyushgarg-dev/piyushgargdev-nextjs#46]
**Describe the bug**
As we take a look at the website,
<img width="656" alt="Screenshot 2023-06-06 180451" src="https://github.com/piyushgarg-dev/piyushgargdev-nextjs/assets/97077594/d11f6d28-b2de-420f-9e18-ce378fdea48a">

 we can see the number of subscribers is not the same as now the number of subscribers as well as the number of YT videos has grown/increased therefore, It should be a good thing to dynamically add the number of subscribers of the YT channel and the number of videos to the website so that the website displays the up-to-date YT channel details (subscribers and no. of YT videos). Also, Hardcoding should be avoided.
<img width="440" alt="Screenshot 2023-06-06 175646" src="https://github.com/piyushgarg-dev/piyushgargdev-nextjs/assets/97077594/dbb31fe6-8362-40e9-95c0-25658de73772">

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by TheMohit2003** [comment]
Is there any youtube API, which lets us know how many subscribers a youtube channel has ?

**Comment by cre8tivestudy** [comment]
Yes, there is a YouTube API that allows us to retrieve information about a YouTube channel, including the number of subscribers. The YouTube Data API provides programmatic access to various aspects of the YouTube platform.

To get the number of subscribers for a specific YouTube channel, we can make use of the channels.list endpoint provided by the YouTube Data API. We will need to obtain an API key from the Google Developers Console and authenticate requests using this key.

Here's an example:

```
import React, { useEffect, useState } from 'react';
import axios from 'axios';

const YouTubeSubscriberCount = ({ channelId, apiKey }) => {
  const [subscriberCount, setSubscriberCount] = useState(0);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const response = await axios.get(
          `https://www.googleapis.com/youtube/v3/channels?part=statistics&id=${channelId}&key=${apiKey}`
        );
        const data = response.data;
        const count = parseInt(data.items[0].statistics.subscriberCount);
        setSubscriberCount(count);
      } catch (error) {
        console.error('Error fetching subscriber count:', error);
      }
    };

    fetchData();
  }, [channelId, apiKey]);

  return (
    <div>
      <h2>Subscriber Count: {subscriberCount}</h2>
    </div>
  );
};

export default YouTubeSubscriberCount;
```

But for this we need Piyush's API key "YouTube Data API (v3)" and Channel ID.
Channel Id can be find easily.

**Comment by TheMohit2003** [comment]
@cre8tivestudy , i just went through a bit of Google search and found out that the channel id can only be accessed by the channel owner. In this case , only piyush knows his channel Id which he can find in the advance setting of his YouTube channel. Hence that part of the code can be written as `process.env.channel_Id` , with channel_Id being a variable.

Coming to the api key, every developer has his own key which he can get from Google developer console. I tried searching for , how to get YouTube api key but found no way . Yes there are blogs and articles but none of them give clear steps on how we can get the api key . Can you find out how we can get the api keh?

**Comment by piyushgarg-dev** [comment]
This issue will be fixed by #15
Thanks to all for contributing, this is an amazing Issue and discussion.

@cre8tivestudy @Satyxm @TheMohit2003
